### PR TITLE
Add tests for dht module

### PR DESCRIPTION
**Pull request by toomyem**
I have added tests for DHT module. Also updated documentation about available modules for esp8266 based boards.

### CODE REVIEW

**Comment by toomyem** [comment]
By the way, I noticed that in config_default.h DHT module is not initialized by default, as it is done with ds18b20 for example. Is it done for purpose of missing by accident?

```
#ifndef CONFIG_MODULE_INIT_DS18B20
#    if CONFIG_DS18B20 == 1
#        define CONFIG_MODULE_INIT_DS18B20                  1
#    else
#        define CONFIG_MODULE_INIT_DS18B20                  0
#    endif
#endif
```

and later in sys.c:

```
static void init_drivers(void)
{
(...)
#if CONFIG_MODULE_INIT_DS18B20 == 1
    ds18b20_module_init();
#endif
(...)
```

File: tst/drivers/software/sensors/dht/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#
+
+NAME = dht_suite
+TYPE = suite
+BOARD ?= linux
+
+CDEFS += \
+	CONFIG_DHT=1 \
+	CONFIG_PIN=1 \
+	PORT_HAS_DHT
```
[the remark this answers is not in the document]
Comment (toomyem) [reply tst/drivers/software/sensors/dht/Makefile]: I was thinking about putting it in config_default.h, but on the other hand linux has no port for pin-related stuff. So it has no dht in fact. That's why I decided to add it for tests only.

File: tst/drivers/software/sensors/dht/Makefile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+STUB = $(addprefix $(SIMBA_ROOT)/src/drivers/sensors/dht.c:, \
+	sys_lock \
+	sys_unlock \
+	pin_port_device_read \
+	time_micros_elapsed \
+	)
```
[the remark this answers is not in the document]
Comment (toomyem) [reply tst/drivers/software/sensors/dht/Makefile]: Ok, I will move paren a line up.

File: tst/drivers/software/sensors/dht/main.c [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+void STUB(sys_lock)(void)
+{
+}
+
+void STUB(sys_unlock)(void)
+{
+}
+
+static void mock_dht_response(int edge_time) {
+    mock_pin_port_device_read(1);
```
[the remark this answers is not in the document]
Comment (toomyem) [reply tst/drivers/software/sensors/dht/main.c]: Ah, I'm just used to java programming style and always forget to put to { on the separate line :)

File: tst/drivers/software/sensors/dht/main.c [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+{
+    int value;
+
+    harness_mock_read("pin_port_device_read(): return(value)",
+                      &value,
+                      sizeof(value));
+    return (value);
+}
+
+static void mock_time_micros_elapsed(int value)
```
[the remark this answers is not in the document]
Comment (toomyem) [reply tst/drivers/software/sensors/dht/main.c]: Sure thing, will do.

File: tst/drivers/software/sensors/dht/main.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (eerimoq) [reply tst/drivers/software/sensors/dht/main.c]: =)

File: doc/boards/esp01.rst [hunk 5]
```diff
@@ -28,6 +28,7 @@ Supported drivers for this board.
 - :doc:`../library-reference/drivers/basic/adc`
 - :doc:`../library-reference/drivers/basic/analog_input_pin`
 - :doc:`../library-reference/drivers/sensors/bmp280`
+- :doc:`../library-reference/drivers/sensors/dht`
```
[the remark this answers is not in the document]
Comment (toomyem) [reply doc/boards/esp01.rst]: I will try to regenerate docs tomorrow.

File: tst/drivers/software/sensors/dht/main.c [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+int STUB(pin_port_device_read)(const struct pin_device_t *dev_p)
+{
+    int value;
+
+    harness_mock_read("pin_port_device_read(): return(value)",
+                      &value,
+                      sizeof(value));
+    return (value);
+}
```
[the remark this answers is not in the document]
Comment (toomyem) [reply tst/drivers/software/sensors/dht/main.c]: Will do in other PR.

File: doc/boards/esp01.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (toomyem) [reply doc/boards/esp01.rst]: Regenerated. Lot of changes, so maybe you could run _make doc_ on your side, just to verify.

[the remark this answers is not in the document]
Comment (eerimoq) [reply doc/boards/esp01.rst]: Looks good.

File: tst/drivers/software/sensors/dht/main.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (eerimoq) [reply tst/drivers/software/sensors/dht/main.c]: Ok.

**Pull request merged**